### PR TITLE
WT-8023 Use global transaction information to asses if a session has active transaction

### PR DESCRIPTION
**Pull request by hbokhari**

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -140,42 +140,43 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
...
+        /* Skip inactive or internal sessions. */
+        if (!session_in_list->active || F_ISSET(session_in_list, WT_SESSION_INTERNAL))
             continue;
-        /* Check if a user session has a running transaction. Ignore prepared transactions. */
-        if (F_ISSET(session_in_list->txn, WT_TXN_RUNNING) &&
-          !F_ISSET(session_in_list, WT_SESSION_INTERNAL) &&
-          !F_ISSET(session_in_list->txn, WT_TXN_PREPARE)) {
 
+        /* Check if a user session has a running transaction. */
+        if (txn_shared->id != WT_TXN_NONE || txn_shared->pinned_id != WT_TXN_NONE) {
```
Comment (hbokhari) [inline src/txn/txn.c:173]: Once a transaction rolls back, commits or get prepared, `__txn_remove_from_global_table()` is called that should clear both `id` and `pinned_id`. As we are holding global transaction lock, no new transaction can start while we are looping.

File: src/txn/txn.c [hunk 2]
```diff
@@ -140,42 +140,43 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
...
+    WT_TXN_SHARED *txn_shared;
     uint32_t i, session_cnt;
     bool txn_active;
 
     conn = S2C(session);
     txn_active = false;
+    txn_global = &conn->txn_global;
+
+    /* We're going to scan the table: wait for the lock. */
+    __wt_readlock(session, &txn_global->rwlock);
```
Comment (kommiharibabu) [inline src/txn/txn.c:161]: The read lock doesn't prevent the transactions that are starting in parallel to the check that is performed here. We need to take the write lock.

Comment (hbokhari) [reply src/txn/txn.c]: Fixed in 8c3a76b

**Review by kommiharibabu — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**